Tella/Scenes/HomeView/Views/FileListView/Add files/PhotoVideoViewModel.swift

```diff
@@ -7,71 +7,169 @@ import Foundation
 import UIKit
 import SwiftUI
 import Photos
+import AVFoundation
+import AVKit
+import AssetsLibrary
 
 class PhotoVideoViewModel : ObservableObject {
     
     var mainAppModel : MainAppModel
     var folderPathArray: [VaultFile] = []
     var  resultFile : Binding<[VaultFile]?>?
-    
+    var  rootFile : Binding<VaultFile?>?
+
     init(mainAppModel: MainAppModel,
          folderPathArray: [VaultFile],
-         resultFile : Binding<[VaultFile]?>? ) {
-        
+         resultFile : Binding<[VaultFile]?>?,
+         rootFile : Binding<VaultFile?>?) {
         self.mainAppModel = mainAppModel
         self.folderPathArray = folderPathArray
         self.resultFile = resultFile
+        self.rootFile = rootFile
     }
-    
-    func add(files: [URL], type: TellaFileType) {
+
+    func handleAddingFile(_ imagePickerCompletion: ImagePickerCompletion?) {
+        let isPreserveMetadataOn = mainAppModel.settings.preserveMetadata
+        if let completion = imagePickerCompletion {
+           
+            switch completion.type {
+            case .video:
+                self.handleAddingVideo(completion, isPreserveMetadataOn)
+            case .image:
+                self.handleAddingImage(completion, isPreserveMetadataOn)
+            }
+        }
+    }
+
+    /// To handle adding the video based on either the user want to preserve the metadata or not
+    /// - Parameters:
+    ///   - completion: Object which contains all the information needed when the user selects a video from Gallery
+    ///   - isPreserveMetadataOn: Flag to check whether the user want to preserve the metadata or not
+    func handleAddingVideo(_ completion: ImagePickerCompletion, _ isPreserveMetadataOn: Bool) {
+        if isPreserveMetadataOn{
+            self.addVideoWithExif(completion)
+        } else {
+            self.addVideoWithoutExif(completion)
```

Comment:
We can also use temporary operator to improve the code     `preserveMetadata ? addVideoWithExif(completion) : addVideoWithoutExif(completion)`

Reply:
Used ternary operator